src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -112,9 +118,64 @@ public Builder toBuilder() {
     private Builder() {
     }
 
+    /**
+     * Adds code using named arguments.
+     *
+     * <p> Named arguments specify their name after the '$' followed by : and the corresponding
+     * type character. For example, to refer to the class {@link java.lang.String} with the argument
+     * name 'text' use a format string containing {@code $text:S} and include the key 'text' with
+     * value {@link java.lang.String} in the argument map.
+     */
+    public Builder addNamed(String format, Map<String, Object> arguments) {
```

Comment:
Probably easier to call than `Map<String, Object>` is `Map<String, ?>`. That way callers can provide a `Map<String, String>` and they won’t get a type error.